clang/lib/Tooling/DependencyScanning/DependencyScanningTool.cpp

```diff
@@ -569,14 +575,103 @@ DependencyScanningTool::getFullDependenciesLegacyDriverCommand(
     LookupModuleOutputCallback LookupModuleOutput,
     std::optional<StringRef> ModuleName) {
   FullDependencyConsumer Consumer(AlreadySeen, LookupModuleOutput,
-                                  Worker.shouldEagerLoadModules());
+                                  Worker.shouldEagerLoadModules(),
+                                  Worker.getCASFS());
   llvm::Error Result =
       Worker.computeDependencies(CWD, CommandLine, Consumer, ModuleName);
   if (Result)
     return std::move(Result);
   return Consumer.getFullDependenciesLegacyDriverCommand(CommandLine);
 }
 
+Error FullDependencyConsumer::initialize(CompilerInstance &ScanInstance,
+                                         CompilerInvocation &CI) {
+  if (CacheFS) {
```

Comment:
nit: reverse the condition to save some indentation.

Reply:
This one I want to keep as-is, because I expect we will need to unify this code with include-tree to support modules, and I think it will reduce the delta.